### PR TITLE
Update developer diagrams

### PR DESCRIPTION
**Pull request by bentsherman**
- Add a new "workflow diagram" for the Nextflow codebase to show the dataflow relationships between classes
- Simplify the class diagrams to focus on structural relationships, should make them easier to read

### CODE REVIEW

File: docs/developer/nextflow.k8s.md [hunk 1]
```diff
@@ -14,4 +14,4 @@ Some classes may be excluded from the above diagram for brevity.
 
 ## Notes
 
-The Kubernetes integration uses the K8s REST API to interact with K8s clusters, and relies on the `kubectl` command and `~/.kube/config` file for authentication.
+The Kubernetes integration uses the K8s HTTP API to interact with K8s clusters, and relies on the `kubectl` command and `~/.kube/config` file for authentication.
```
Comment (pditommaso) [inline docs/developer/nextflow.k8s.md:17]: `REST API` is fine https://kubernetes.io/docs/reference/using-api/

Comment (bentsherman) [reply docs/developer/nextflow.k8s.md:17]: REST is overused, most APIs that claim to be REST APIs are not actually REST

Comment (bentsherman) [reply docs/developer/nextflow.k8s.md:17]: See this post from the creator of REST: https://roy.gbiv.com/untangled/2008/rest-apis-must-be-hypertext-driven

It is a minor point but I'm trying to not perpetuate the error

Comment (pditommaso) [reply docs/developer/nextflow.k8s.md:17]: Fair enough

**Pull request merged**